src/base/http/requestparser.h

```diff
@@ -37,32 +38,35 @@ namespace Http
     class RequestParser
     {
     public:
-        enum ErrorCode
+        enum class ParseStatus
         {
-            NoError = 0,
-            IncompleteRequest,
+            OK,
+            Incomplete,
             BadRequest
         };
 
-        // when result != NoError parsed request is undefined
-        // Warning! Header names are converted to lower-case.
-        static ErrorCode parse(const QByteArray &data, Request &request, uint maxContentLength = 10000000 /* ~10MB */);
+        struct ParseResult
+        {
+            // when `status != ParseStatus::OK`, `request` & `frameSize` are undefined
+            ParseStatus status;
+            Request request;
+            long frameSize;  // http request frame size (bytes)
+        };
+
+        static ParseResult parse(const QByteArray &data);
+
+        static const long MAX_CONTENT_SIZE = 64 * 1024 * 1024;  // 64 MB
 
     private:
-        RequestParser(uint maxContentLength);
```

Comment:
The idea is RequestParser is used for different web applications (WebUI and embedded Tracker) that can require different limits on content length.

Reply:
I didn't see other application than webUI using `RequestParser` ... maybe change this in later PR?

Reply:
I don't mind. Besides, I don't remember how exactly it was designed right now.